### PR TITLE
Add Calculated Attribute Definition resource

### PR DESCRIPTION
**Pull request by will-hodge**
Adding the new recently-launched resource Calculated Attribute Definition.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-customerprofiles-calculatedattributedefinition.html

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-customerprofiles-calculatedattributedefinition/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+Congratulations on starting development! Next steps:
+
+1. Write the JSON schema describing your resource, `aws-customerprofiles-calculatedattributedefinition.json`
+1. Implement your resource handlers.
+
+The RPDK will automatically generate the correct resource model from the schema whenever the project is built via Maven. You can also do this manually with the following command: `cfn generate`.
+
+> Please don't modify files under `target/generated-sources/rpdk`, as they will be automatically overwritten.
+
+The code uses [Lombok](https://projectlombok.org/), and [you may have to install IDE integrations](https://projectlombok.org/setup/overview) to enable auto-complete for Lombok-annotated classes.
```
Comment (ynwangn) [inline aws-customerprofiles-calculatedattributedefinition/README.md:12]: Are these needed? Or a better question, should we modify this to be more public friendly? Like provide some basic guidance on how to use and links to related resources?

File: aws-customerprofiles-calculatedattributedefinition/lombok.config [hunk 2]
```diff
@@ -0,0 +1 @@
+lombok.addLombokGeneratedAnnotation = true
```
Comment (ynwangn) [inline aws-customerprofiles-calculatedattributedefinition/lombok.config:1]: Is this needed externally?

File: aws-customerprofiles-calculatedattributedefinition/src/resources/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+        </File>
+    </Appenders>
+    <Loggers>
+        <Root level="DEBUG">
+            <AppenderRef ref="Console"/>
+            <AppenderRef ref="APPLICATION"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (ynwangn) [inline aws-customerprofiles-calculatedattributedefinition/src/resources/log4j2.xml:17]: This is not needed externally.

**Review by ynwangn — CHANGES_REQUESTED** [review]
Please inspect for other content that might not be useful to the public.

File: aws-customerprofiles-calculatedattributedefinition/src/resources/log4j2.xml (hunk 3 above)
Comment (vallabja) [reply aws-customerprofiles-calculatedattributedefinition/src/resources/log4j2.xml:17]: Agree, will remove src/resources

File: aws-customerprofiles-calculatedattributedefinition/lombok.config (hunk 2 above)
Comment (vallabja) [reply aws-customerprofiles-calculatedattributedefinition/lombok.config:1]: I think yes, looking at the existing providers repositories, all of them have this config file present.

File: aws-customerprofiles-calculatedattributedefinition/README.md (hunk 1 above)
Comment (vallabja) [reply aws-customerprofiles-calculatedattributedefinition/README.md]: Addressed, made this file public friendly.

**Review by ynwangn — APPROVED** [review]

**Review by lanxin-zhang-work — APPROVED** [review]

**Pull request merged**